### PR TITLE
Center selected thumb in Filmstrip #5174

### PR DESCRIPTION
**Pull request by Beep6581**
When opening an image, selecting the next/previous image or syncing the
Filmstrip, the newly selected thumbnail is centered, unless it lies
within the visible area more than 1 thumbnails-width away from either
edge, in which case centering does not occur.

### CODE REVIEW

File: rtgui/filebrowser.cc [hunk 1]
```diff
@@ -1727,62 +1727,62 @@ void FileBrowser::buttonPressed (LWButton* button, int actionCode, void* actionD
...
+                            if ((x2 > x1 + ww - 2 * tw) || (x2 - tw < x1)) {
+                                setScrollPosition(x2 - (ww - tw) / 2, y2);
                             }
 
                             // open the selected image
                             std::vector<Thumbnail*> entries;
-                            entries.push_back (thumb);
-                            tbl->openRequested (entries);
+                            entries.push_back(thumb);
+                            tbl->openRequested(entries);
```
Comment (Floessie) [inline rtgui/filebrowser.cc]: If I get that right from the diff, these three lines could be replaced by one:
```c++
tbl->openRequested({thumb});
```

File: rtgui/filebrowser.cc [hunk 2]
```diff
@@ -1792,62 +1792,62 @@ void FileBrowser::openNextImage ()
...
```
Comment (Floessie) [inline rtgui/filebrowser.cc]: Same here.

File: rtgui/filebrowser.h [hunk 3]
```diff
@@ -181,12 +181,13 @@ class FileBrowser  : public ThumbBrowserBase,
         return tbl ? tbl->isInTabMode() : false;
     }
 
-    void openNextImage ();
-    void openPrevImage ();
+    void openNextImage();
+    void openPrevImage();
+    void selectImage(Glib::ustring fname);
```
Comment (Floessie) [inline rtgui/filebrowser.h]: This should really be `void selectImage(const Glib::ustring& fname)`.

File: rtgui/filebrowser.cc (hunk 2 above)
Comment (Floessie) [reply rtgui/filebrowser.cc]: Hm, looks like a lot could be factored out here...

File: rtgui/filebrowser.h [hunk 4]
```diff
@@ -181,12 +181,13 @@ class FileBrowser  : public ThumbBrowserBase,
...
+
     void copyProfile ();
     void pasteProfile ();
     void partPasteProfile ();
-    void selectImage (Glib::ustring fname);
     void openNextPreviousEditorImage (Glib::ustring fname, eRTNav eNextPrevious);
```
Comment (TooWaBoo) [inline rtgui/filebrowser.h]: @Floessie 
...then this should be `void openNextPreviousEditorImage (const Glib::ustring fname, eRTNav eNextPrevious);` too

**Comment by TooWaBoo** [comment]
The behavior between` F3/F4` and `<shift> F3/F4` or buttons `Left/Right` is different.

`F3/F4` centers the image on the last fully visible thumb.

 `<shift> F3/F4` or buttons `Left/Right` always scrolls the filmstrip and the selcted thumb is always one thumb left or right from the center.

**Comment by TooWaBoo** [comment]
This will fix the navigatopn issue
### filebrowser.cc
old
```
void FileBrowser::openNextPreviousEditorImage (Glib::ustring fname, eRTNav nextPrevious)
{
    // let FileBrowser acquire Editor's perspective
    selectImage (fname);

    // now switch to the requested image
    if (nextPrevious == NAV_NEXT) {
        openNextImage();
    } else if (nextPrevious == NAV_PREVIOUS) {
        openPrevImage();
    }
}
```

new
```
void FileBrowser::openNextPreviousEditorImage (Glib::ustring fname, eRTNav nextPrevious)
{
    // switch to the requested image
    if (nextPrevious == NAV_NEXT) {
        openNextImage();
    } else if (nextPrevious == NAV_PREVIOUS) {
        openPrevImage();
    } else {
        // let FileBrowser acquire Editor's perspective
        selectImage (fname);
    }
}
```

File: rtgui/filebrowser.cc (hunk 1 above)
Comment (Beep6581) [reply rtgui/filebrowser.cc]: Thank you @Floessie , I will do that.

**Comment by Beep6581** [comment]
Thank you @TooWaBoo , I wasn't aware of the `<shift>` variant, will look into that. What do you mean by "buttons Left/Right", do you mean the   buttons below the main preview?

**Comment by TooWaBoo** [comment]
> What do you mean by "buttons Left/Right", do you mean the   buttons below the main preview?

Yes, it's the same function as `<shift> F3/F4`

**Comment by Beep6581** [comment]
I believe the branch is ready for merging. @TooWaBoo could you please test again?

**Review by Floessie — COMMENTED** [review]
> Default to using defaults by default

:rofl:

**Comment by TooWaBoo** [comment]
> 
> 
> I believe the branch is ready for merging. @TooWaBoo could you please test again?

Tested and it works fine.

**Pull request merged**